application/src/main/java/org/thingsboard/server/service/install/update/DefaultDataUpdateService.java

```diff
@@ -209,116 +218,54 @@ public void updateData(String fromVersion) throws Exception {
-                        case "org.thingsboard.rule.engine.metadata.TbGetDeviceAttrNode":
-                            fetchTo = checkEnrichmentNodeFetchProperty(configObjectNode, "fetchToData", FetchTo.DATA, FetchTo.METADATA);
-                            break;
-                        case "org.thingsboard.rule.engine.metadata.TbGetCustomerDetailsNode":
-                        case "org.thingsboard.rule.engine.metadata.TbGetTenantDetailsNode":
-                            fetchTo = checkEnrichmentNodeFetchProperty(configObjectNode, "addToMetadata", FetchTo.METADATA, FetchTo.DATA);
-                            break;
-                        case "org.thingsboard.rule.engine.metadata.TbFetchDeviceCredentialsNode":
-                            fetchTo = checkEnrichmentNodeFetchProperty(configObjectNode, "fetchToMetadata", FetchTo.METADATA, FetchTo.DATA);
-                            break;
-                        case "org.thingsboard.rule.engine.metadata.TbGetOriginatorFieldsNode":
-                        case "org.thingsboard.rule.engine.metadata.TbGetRelatedAttributeNode":
-                        case "org.thingsboard.rule.engine.metadata.TbGetTenantAttributeNode":
-                        case "org.thingsboard.rule.engine.metadata.TbGetCustomerAttributeNode":
-                            fetchTo = FetchTo.METADATA;
-                            break;
-                        default:
-                            log.error("Failed to update rule node: [{}] with id: [{}] " +
-                                    "Reason: Unexpected rule node type!", ruleNodeType, ruleNode.getId());
-                            continue;
-                    }
-
-                    if (fetchTo == null) {
-                        log.error("Failed to update rule node: [{}] with id: [{}]", ruleNodeType, ruleNode.getId());
-                        continue;
-                    }
-
-                    configObjectNode.put("fetchTo", fetchTo.name());
-                    ruleNode.setConfiguration(configObjectNode);
-                    var ruleChainId = ruleNode.getRuleChainId();
-                    var tenantId = ruleChainIdToTenantId.computeIfAbsent(ruleChainId,
-                            id -> {
-                                RuleChain ruleChain = ruleChainService.findRuleChainById(TenantId.SYS_TENANT_ID, id);
-                                if (ruleChain == null) {
-                                    log.error("Failed to find rule chain by id: [{}], ruleNodeId: [{}]", ruleChainId, ruleNode.getId());
-                                    return null;
-                                }
-                                return ruleChain.getTenantId();
-                            });
-                    if (tenantId != null) {
-                        ruleChainService.saveRuleNode(tenantId, ruleNode);
-                    }
-                }
-            });
+            upgradeRuleNode(ruleChainIdToTenantId, new TbGetOriginatorFieldsNode());
```

Comment:
Let's write this code to be generic and once and for all cases:
1) add RuleNode.configurationVersion (0 by default)
2) Iterate through all classes that has RuleNode annotation
3) Check that the class is an instance of VersionedNode
4) Check findAllRuleNodesByTypeAndVersionLessThen(versionedNode.getClass().getName(), versionedNode.getCurrentVersion(), pageLink)
5) update them.

Reply:
6. add good logging: Rule Node Id, fromVersion, toVersion,  etc.